### DISCUSSION TITLE
How to use quarto environment variables from _environment as if in .Renviron in an R script?

### DISCUSSION THREAD

**Discussion by maia-sh** [quarto-dev/quarto-cli#8975] [Q&A] [answered]
### Description

The issue: I have global variables I would like to use in both the Quarto document as well as R scripts *not* sourced within the Quarto document (i.e., they are sourced prior for expensive computations, which I prefer not to cache or freeze.) However, I can't **access the Quarto environment variables from the separate R scripts within the project**.

I have created an `_environment` file per [Quarto Environment Variables documentation](https://quarto.org/docs/projects/environment.html). For example:
````
HTTP_PROXY="https://proxy.example.com/"
````

I am able to use the variable within the Quarto document using a shortcode (i.e., `{{< env HTTP_PROXY >}}`) as well as in R chunks (i.e., `Sys.getenv("HTTP_PROXY")`).

I would like use the same variable in separate R scripts (e.g., also using `Sys.getenv("HTTP_PROXY")`) and avoid duplicating the same environment variables to `.Renviron` file per [R documentation](https://support.posit.co/hc/en-us/articles/360047157094-Managing-R-with-Rprofile-Renviron-Rprofile-site-Renviron-site-rsession-conf-and-repos-conf)

Any ideas for a workaround to use Quarto environment variables from the separate R scripts within the project? Ideally, I can set `.Renviron` to instead read from `_environment`

Thanks for the help!

**Comment by mcanouil** [comment]
This is not really a Quarto question.

Quarto environment variables (`_environment`) is for Quarto.

If you want to use environment variables more broadly, then it defers to your language and operating system tooling, so `.Renviron` would work for R but only for R while bash/shell would work for all.

Note that you can also read `_environment` from R, parse the file and assign environment variables from R, for R to use.

**Reply by cderv** [reply, marked as answer]
I believe the `_environment` file as a format compatible with R startup file `.Renviron`, so probably doing `base::readRenviron("_environment")` works to load the file content as environment variable. 

Otherwise, the file syntax is simple enough to do a simple R function to read each line, split the lines and set the name value for each environment variable. 

Also Currently `quarto run` does not load the `_environment` file - this is only done in Quarto project context. We could maybe imagine a feature request for an option to load environment files context with `quarto run`.
